third_party/rust/sha2/v0_9/BUILD.gn

```diff
@@ -46,11 +46,8 @@ cargo_crate("lib") {
     "//brave/third_party/rust/digest/v0_9:lib",
     "//brave/third_party/rust/opaque_debug/v0_3:lib",
   ]
-  if (target_cpu == "arm64" || target_cpu == "x64" || target_cpu == "x86") {
+  if (current_cpu == "arm64" || current_cpu == "x64" || current_cpu == "x86") {
     deps += [ "//brave/third_party/rust/cpufeatures/v0_2:lib" ]
   }
   features = [ "std" ]
-  if (target_os == "android" && target_cpu == "arm64") {
```

Comment:
I believe this is going to break things because cpufeatures does not support android

Reply:
cc: @cdesouza-chromium @AlexeyBarabash

Reply:
I guess if it's building then it's probably ok, but we added this specifically because it broke in android. Maybe it's only for some build types?

Reply:
Interesting that gnrt is using `current_cpu` here when the Cargo.toml has `target_cpu`. I think that would unnecessarily add the dep for build scripts in some cases, but probably won't hurt anything. I'll check in chromium slack

Reply:
more details what error happened and required the fix is at 
https://bravesoftware.slack.com/archives/CU5UXM3EC/p1726785702571649